### ISSUE TITLE
Long options .jl string comparisons

### ISSUE THREAD

**Issue by rbdavis** [PaulWessel/CodingTasks#4]
Hi Paul, now going through all of the string choices again in every module we've done, comparing to the .jl strings.

What do you think about using camelCase for these strings? I'm working in alphabetic order of module name, starting with blockmean. There, I've discovered hyphens in your version, and underscores in the .jl version (in different places). This is kind of a mess, I think we should at least start with a common style and decide on possible exceptions as needed.

**Comment by PaulWessel** [comment]
I think blockmean w/ common options was the first I did and that predates the discussion with the wrappers on not using hyphens... Unfortunately, we did not reach a decision on what system to use at the meeting on THursday.   I think for sure we will dispense with the hyphens.  As for the other ones I am not sure what to tell you.  I think if we decide later to go with, say, Camel Case then Joaquim will add aliases to cover those plus his original ones (he kind of has to do that to remain backward compatible).    Since I cannot make a unilateral decision on the format at this point (I can try to get this accelerated this coming 4 weeks) but I am busy packing right now for an early night and an early flight out tomorrow (to NYC) and then wont be working much while in NYC helping Malia get organized.  So unlikely to do much until I am in Hawaii end of the month and then I think you are doing cruise stuff and off on vacation.

Since it is not that many modules yet I think a reasonable measure right now is to match Julia unless there are clear english-language violation (he is Portuguese so that can happen) in which case you can stick with what you have for now.

**Comment by rbdavis** [comment]
OK, I will  eyeball all the modules in Julia first to make sure there is at least consistency with underscore usage there. If so, I will duplicate. Not a big underscore fan (e.g., n_points, etc., looks dumb to me, or do you just make lots of exceptions and do npoints, etc.?), but will not rock the boat.

Shouldn't take me too long to get through these pre-longopted modules this time around, so if you have any ideas on next modules to do please advise, otherwise I'll just pick my own at random or whatever.

Have a good time in NYC, hope it's not too hot while you're there! (Late August, what are the chances? ;-> )

**Comment by rbdavis** [comment]
The .jl strings seem to be a mixture of underscores and runonwords. I guess I'll try to favor the latter style where it remains intelligible, otherwise use the underscores.

**Comment by rbdavis** [comment]
I'm thinking maybe it does not make sense for me to submit any more PRs in the next few days with you being on travel? I will probably be making almost no changes other than editing these translation tables, so I can easily just squirrel away my own copies of the new tables and copy them into the current pulled modules whenever you've got time to get back to the PR process. Please advise if you'd like me to do otherwise!

**Comment by PaulWessel** [comment]
I think it is OK for you to submit single module PRs.  I will have time each day to review etc.  That way it does not build up so much.  OK, checking out now!

**Comment by rbdavis** [comment]
Have a good trip!

**Comment by rbdavis** [comment]
Hi Paul, hope you had (or are about to have) a good flight to NYC!

OK, one measly day going down the path of doing one module per PR and I'm already having a hard time with git doing this.

(1) I created a new branch to modify the blockmean longopts, changed the code, tested it (redoing the early modules you worked on also involves long-optmangling the test scripts, so that took a little more time than expected, but no big deal), then published the branch and created a PR. So far, so good.

(2) Now I create a second new branch to do the same for blockmedian. I change the code, test, all good, and now I run afoul of git. When I publish this new blockmedian branch and start to create a second PR from that, it wants to also include the changed blockmean.c file in the PR. How do I get from where I am to a state where only the changed blockmedian.c file is included in the PR?

Until someone can tell me how to fix this mess, I'm just going to keep working on the various modules but squirrel away copies of the new code in a safe place on my local system. I suppose as you approve each PR I can then upload the next change and wait for approval on that, etc.

**Comment by PaulWessel** [comment]
After any pull request you should go back and check out master before you start on another branch. Otherwise all the changes you made are still in the branch you started with.

**Comment by rbdavis** [comment]
Thanks, I'll keep that in mind for next time!

**Comment by rbdavis** [comment]
> After any pull request you should go back and check out master before you start on another branch. Otherwise all the changes you made are still in the branch you started with.

Actually no, this does not work, at least not for me. And I have tried this before, I should have recalled that last night before I replied. It seems that when I change branch back to master and do a pull, it refuses to revert any file that is newer than what's in the master (at least that's my guess). I've actually spent time googling how to re-download a specific file from the origin's master branch to override this behavior, but the answers are confusing and complex and, like doing anything else with git, I'm concerned there's a risk I will trash something.

I just tried it again this morning and here's what happened:

```
>  git checkout master
M	.gitignore
M	cmake/ConfigUserAdvancedTemplate.cmake
M	src/blockmode.c
Switched to branch 'master'
Your branch is up to date with 'origin/master'.

>  git pull
Already up to date.

>  ls -l block*c
64 -rw-r--r--  1 rbd  staff  31521 Aug 21 06:24 blockmean.c
72 -rw-r--r--  1 rbd  staff  34660 Aug 21 06:24 blockmedian.c
88 -rw-r--r--  1 rbd  staff  42311 Aug 21 06:20 blockmode.c
```

The timestamps shown in the final ls -l above indicate that all the files I was working on yesterday, and which have not yet been merged into the origin's master, remain in their modified version, and were not replaced by older files.

The checkout comment notices that my local copy of blockmode.c is modified ( **M** ?), I'm not sure why it does not notice the other two modified block*c files, but in any event the checkout+pull does not revert any of them back to their state prior to yesterday's work.

**Comment by PaulWessel** [comment]
Hi Roger, now in NYC.  I am not sure what is going on.  I am using GitHub Desktop exclusively for these things so I do not actually know the command line equivalents without googling around.  But at least there, when I complete a pull request and switch to master I am getting master and not any trace of things I just pushed in the PR.

**Comment by rbdavis** [comment]
Hi Paul, I tried again after removing some files and it seems to have worked this time. I'm really not at all sure what happened earlier, but things seem OK git-wise for the moment so let's just assume I did something stupid.

**Comment by PaulWessel** [comment]
Sure, no problem.  If you truly get stuck with a git command sequence then best to ask for help from Dongdong or Max via the `@name` system here, since I would just be an intermediate google person.

**Comment by rbdavis** [comment]
Hope your visit with Malia is going well! I haven't been to The City in a VERY long time.

Thanks for being so prompt with these PR approvals, but I hope you aren't being unduly distracted by them from your visit. I noticed that the check after your grdimage PR approval failed on the Windows build (while compiling a file I was miles away from touching, gmt_something.c!) Does this restart itself at some point? It seems to have given up as far as I can make out. Only asking because it appears I can't squash/merge until the mess is unmessed. Please don't prioritize this, I'll just keep working on other modules in any case!

**Comment by PaulWessel** [comment]
I'm trying to rerun that compilation.  Looks like an internal crash in the cloud unrelated to GMT code per se.  Sometimes there are failures in the cloud.  You can click on them and see - if they have trouble downloading prereqs or access remote sites then it is OK to just overwrite and merge/squash.  Let's see if it completes or fails the same way.

**Comment by rbdavis** [comment]
Looks like all good now, thanks!

**Comment by rbdavis** [comment]
> Hi Roger, here we will do a few things differently:
> 
>  1. -F: Use interpolator instead of interptype.

FYI, I only changed that one to match GMT.jl.

Just a heads-up, I'm trying to match GMT.jl string choices wherever it seems reasonable to do so, but not everywhere, i.e., not in places where I feel more strongly that the GMT.jl choices are not so great. I can imagine there may be some disagreement on these. I would estimate that so far I've probably changed strings to match in maybe 60% of the cases where there are disagreements. (Probably the strings were already in agreement about 50% or more of the time.) I'm always happy to explain my reasoning on any particular choice to anyone who cares, but wasn't sure if it was a great idea to do that in detail in the PRs.

It does seem that a large percentage of the GMT.jl strings already have aliases (often a few aliases per string), so in cases where no agreement can be reached on which strings are better, another alias could always be added in GMT.jl to bring the two string sets into better alignment.

**Comment by PaulWessel** [comment]
Yes, I understand.  And agreed that Joaquim can always add an alias.

**Comment by rbdavis** [comment]
Wow, ugly stuff going on in the Windows builds of the last 3 PRs you approved. I assume they are all hung up in the same place?

**Comment by PaulWessel** [comment]
Yes, I would just ignore and squash since there is nothing windows-specific we need to check (i.e., no #ifdef WIN32 etc that would compile things differently for Windows)

**Comment by rbdavis** [comment]
OK, so just check the box that says 'Merge without waiting for requirements to be met ...' , and then proceed?

**Comment by PaulWessel** [comment]
Yep.

**Comment by rbdavis** [comment]
Will finish catching up on GMT.jl string comparisons today. Tomorrow will likely be my last GMT day for a while as discussed earlier. Do you have any particular new modules you want me to longopt? Or I could peek around in the translator again and try see if abbreviations of longopt strings could be made to work easily (or not)?

**Comment by rbdavis** [comment]
> Will finish catching up on GMT.jl string comparisons today. Tomorrow will likely be my last GMT day for a while as discussed earlier. Do you have any particular new modules you want me to longopt? Or I could peek around in the translator again and try see if abbreviations of longopt strings could be made to work easily (or not)?

Actually, I guess we already discussed me finishing up the seis modules so I will proceed with that for now.

**Comment by rbdavis** [comment]
Ooops?!?

I was replying to your comment about pyGMT in today's pssac PR and something weird happened. I don't know if I somehow clicked the self-destruct button (pretty sure I didn't), but all of a sudden the webpage redraws, tells me the PR is closed, and I haven't merged anything. WTF?

**Comment by PaulWessel** [comment]
It says you closed it - probably by accident.

**Comment by rbdavis** [comment]
I guess the real question is how do we re-open it? Should I just submit a new PR?

**Comment by PaulWessel** [comment]
See if you can get to it via the commit address?

**Comment by rbdavis** [comment]
Sorry, I don't understand what you mean by that.

**Comment by PaulWessel** [comment]
It gies this link as what you are trying to do (but then you closed it):
https://github.com/GenericMappingTools/gmt/tree/220823-pssac-longopts

**Comment by PaulWessel** [comment]
Maybe you can at least get the changes from there, or just create a new branch.

**Comment by rbdavis** [comment]
No worries, I save changes locally. Shall I just open another PR?

**Comment by rbdavis** [comment]
New pssac branch published and new PR requested, sorry for the mess. I really don't know what I could have clicked on, I thought I was just replying to the message you wrote in the PR. Will look at that page carefully again this time around and see if I can figure out what I might have done.

**Comment by rbdavis** [comment]
OK, I think I see what must have happened. Down in the area where you can enter a comment on the PR page, just below where you type the comment there are two buttons right beside each other -- the rightmost one (in green) is labelled **Comment** , and the one hard up against its left edge is labelled (you got it!) **Close pull request** . That is some seriously awful UI design, I'm sorry. I must have tried to click the Comment button after I finished composing the comment but a smidgen of stray mouse movement did the evil deed. Now that I look at this Issues page where I'm typing this message I see the very same thoughtful layout here, so be careful if you respond to this remark!

**Comment by PaulWessel** [comment]
No worries!

**Comment by rbdavis** [comment]
LMK whenever's good for you to get together during your stay out here. I'm flying out on the 8th, returning late on the 25th. I'll be working most days at the marine center but schedule is flexible and I can come up to campus anytime, or if you'd rather come by the marine center we could do that.

**Comment by PaulWessel** [comment]
Yep, I will see what my schedule will look like once I am back on Saturday and let you know.

**Comment by PaulWessel** [comment]
Hi Roger, hectic on my end.  Would Wednesday 7th at UH work for you?  I expect to be in all day that day.  Only have a lunch with Garrett et a noon-1pm.

**Comment by rbdavis** [comment]
Hi Paul,

Next Wednesday is fine. We leave Thursday night. If you are at Manoa all 
day then I'd say the earlier the better for me as I could stop in on the 
way to the marine center. Or conversely, late in the day so I can stop on 
the way back. Push comes to shove, though, I can come by pretty much 
anytime as necessary.

Roger

**Comment by PaulWessel** [comment]
Hi Roger.  On the GMT zoom we discussed pre-filling the remaining modules with a dummy structure that does nothing, e.g.,  

```
static struct GMT_KEYWORD_DICTIONARY module_kw[] = { /* Local options for this module */
	/* separator, short_option, long_option,
	          short_directives,    long_directives,
	          short_modifiers,     long_modifiers */
	{ 0, 'A', "",
	          "",           "",
	          "",           "" },
	 ...
```

where the only thing filled in is the letter option (here A) but no long option text etc.  I think I can write a script to create that dummy structure and then add it to the modules.  However, I think I need to make a change in the _gmtinit_find_kw_ function to handle a blank long option name.  Perhaps

```
for (*k = 0; kw[set][*k].long_option[0]; (*k)++)
	if (!strcmp (arg, kw[set][*k].long_option)) break;	/* Match was found */
if (kw[set][*k].short_option) return kw[set];	/* Return if successful, otherwise short_option of last entry is blank */

```
needs to be

```
for (*k = 0; kw[set][*k].long_option[0]; (*k)++)
	if (kw[set][*k].long_option[0] && !strcmp (arg, kw[set][*k].long_option)) break;	/* Match was found */
if (kw[set][*k].short_option) return kw[set];	/* Return if successful, otherwise short_option of last entry is blank */
```

That way we only do the string comparison if there is a string.  The idea is to make it as simple as possible for volunteers to add in text.

**Comment by rbdavis** [comment]
Any particular reason for wanting to do this?

BTW, I was having some general thoughts about longopts stuff over the 
weekend while catching up on yard work, figured we could discuss on 
Wednesday.

**Comment by PaulWessel** [comment]
The reason was that when we add empty structures to all the remaining modules that the code will function just fine and skip those sections (i.e., loop over the structure items but not find any long-option argument.
